src/engine/effects/engineeffectchain.cpp

```diff
@@ -128,113 +141,155 @@ bool EngineEffectChain::processEffectsRequest(const EffectsRequest& message,
     return true;
 }
 
-bool EngineEffectChain::enableForChannel(const ChannelHandle& handle) {
-    ChannelStatus& status = getChannelStatus(handle);
-    if (status.enable_state != EffectProcessor::ENABLED) {
-        status.enable_state = EffectProcessor::ENABLING;
+bool EngineEffectChain::enableForInputChannel(const ChannelHandle& inputHandle,
+        const QList<EffectStatesPointer>* statesForEffectsInChain) {
+    qDebug() << "EngineEffectChain::enableForInputChannel" << this << inputHandle;
+    auto& outputMap = m_chainStatusForChannelMatrix[inputHandle];
+    for (auto&& outputChannelStatus : outputMap) {
+        if (outputChannelStatus.enable_state != EffectEnableState::Enabled) {
+            outputChannelStatus.enable_state = EffectEnableState::Enabling;
```

Comment:
If you decide to use the new state in any case, you need here Enabling in any case. Else we need to stick with the old states, if any.

Reply:
Here, we need to have Disabled, for normal operation. All other states are a bit problematic, but we should be aware of it.

Reply:
I think it would make sense to check if the chain is Enabled first, and if it is, delete the states passed in the second parameter to this function and return.

Reply:
Yes, this is a delete in the main thread, but that should not happen in normal conditons.